Source/Redemption/Private/PuzzleLever.cpp

```diff
@@ -0,0 +1,18 @@
+// Fill out your copyright notice in the Description page of Project Settings.
+
+
+#include "PuzzleLever.h"
+#include "InteractWidgetComponent.h"
+
+APuzzleLever::APuzzleLever()
+{
+	InteractComp = CreateDefaultSubobject<UInteractWidgetComponent>(TEXT("Interact Comp"));
+}
+
+void APuzzleLever::Activate()
+{
+	UE_LOG(LogTemp, Warning, TEXT("ACTIVATE!!!!"))
+	IsActive = !IsActive;
+
+	OnActivate.Broadcast();
```

Comment:
What is this broadcast function?

Reply:
OnActivate is a event like OnTakeAnyDamage or OnComponentBeginOverlap.
broadcast is the function that tells all the ".Adddynamic" that it is time for their function to be called

Reply:
thank you